rust/actyx/ax-aql/src/language/parser.rs

```diff
@@ -721,6 +722,116 @@ pub(crate) fn query_from_str(s: &str) -> Result<Query<'_>> {
     Ok(query)
 }
 
+pub(crate) struct QueryWorkflowAnalysisError<'a> {
+    pub(crate) span: Span<'a, Ident>,
+    pub(crate) error_type: QueryWorkflowAnalysisErrorType,
+}
+
+pub(crate) enum QueryWorkflowAnalysisErrorType {
+    UndeclaredEvent,
+    UndeclaredWorkflowCall,
+}
+
+pub(crate) struct QueryWorkflowAnalysis<'a> {
+    pub(crate) query: &'a Query<'a>,
+    pub(crate) workflow_tracker: WorkflowTracker<'a>,
```

Comment:
I think neither of these fields are needed: the (recursive) analysis functions can just get the outer `Query` and the syntax element in question as arguments. When analysing the flow inside a workflow we’ll probably need to have some more context, though, so it might be appropriate to keep this struct with the `query` field in any case.

Reply:
I would also imagine the type checker information to be stored in this struct too because it will be used inside while checking binders